### PR TITLE
Make addrundata write metadata.textproto files

### PR DESCRIPTION
**Pull request by greg-dennis**

### CODE REVIEW

File: tools/addrundata/README.md [hunk 1]
```diff
@@ -21,23 +21,19 @@ There are two modes of operation:
...
-An example `rundata_test.go` looks like this:
+An example `metadata.textproto` looks like this:
 
-```go
-// Code generated by go run tools/addrundata; DO NOT EDIT.
-package foo_test
-
-import "github.com/openconfig/featureprofiles/internal/rundata"
+```
+# proto-file: proto/metadata.proto
```
Comment (liulk) [inline tools/addrundata/README.md:27]: Should we fully-qualify the proto file with repo?

github.com/openconfig/featureprofiles/proto/metadata.proto

File: tools/addrundata/case.go [hunk 2]
```diff
@@ -16,34 +15,26 @@ import (
...
-			return nil
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("could not parse rundata_test.go: %w", err)
 		}
-		return fmt.Errorf("could not parse rundata_test.go: %w", err)
+	} else {
+		tc.deprecated = true
+	}
+	if err := readFile(filepath.Join(testdir, "metadata.textproto"), tc.existing.fromProto); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("could not parse metadata.textproto: %w", err)
```
Comment (liulk) [inline tools/addrundata/case.go:37]: Break a long line?

```suggestion
	err := readFile(filepath.Join(testdir, "metadata.textproto"), tc.existing.fromProto)
	if err != nil && !os.IsNotExist(err) {
		return fmt.Errorf("could not parse metadata.textproto: %w", err)
```

File: tools/addrundata/case.go [hunk 3]
```diff
@@ -16,34 +15,26 @@ import (
 
 // testcase carries parsed rundata from different sources to be fixed and checked.
 type testcase struct {
-	pkg      string     // Package used in the test code.
-	markdown parsedData // From the README.md.
-	existing parsedData // From existing source code.
-	fixed    parsedData // Fixed rundata to write back, populated by fix().
+	markdown   parsedData // From the README.md.
+	existing   parsedData // From existing source code.
```
Comment (liulk) [inline tools/addrundata/case.go:19]:
```suggestion
	existing   parsedData // From existing metadata or deprecated rundata_test.go.
```

File: tools/addrundata/main.go [hunk 4]
```diff
@@ -29,6 +29,9 @@ var (
 	fix       = flag.Bool("fix", false, "Update the rundata in tests.  If false, only check if the tests have the most recent rundata.")
 	list      = flag.String("list", "", "List the tests in one of the following formats: csv, json")
 	mergejson = flag.String("mergejson", "", "Merge the JSON listing from this JSON file.")
+
+	// TODO(greg-dennis): Remove once all rundata_test.go files are replaced.
+	checkDeprecated = flag.Bool("check_deprecated", false, "whether to check for the existence of deprecated rundata_test.go files")
```
Comment (liulk) [inline tools/addrundata/main.go:34]: checkDeprecated -> deprecateRundataTestGo?

File: tools/addrundata/rundata.go [hunk 5]
```diff
@@ -44,10 +47,12 @@ func (pd *parsedData) fromMarkdown(r io.Reader) error {
 // fromCode reads parsedData from a source code.
 func (pd *parsedData) fromCode(r io.Reader) error {
 	sc := bufio.NewScanner(r)
+	var foundInit bool
```
Comment (liulk) [inline tools/addrundata/rundata.go:50]: Don't need this?

File: tools/addrundata/rundata.go [hunk 6]
```diff
@@ -56,6 +61,32 @@ func (pd *parsedData) fromCode(r io.Reader) error {
...
+	}
+	if uuid := md.GetUuid(); uuid != "" {
+		pd.testUUID = uuid
+	}
+	if planID := md.GetPlanId(); planID != "" {
+		pd.testPlanID = planID
+	}
+	if desc := md.GetDescription(); desc != "" {
+		pd.testDescription = desc
+	}
```
Comment (liulk) [inline tools/addrundata/rundata.go:88]: This initializes pd, so we don't need to care about setting fields conditionally.

```suggestion
	pd.testUUID = md.GetUuid()
	pd.testPlanID = md.GetPlanId()
	pd.testDescription = md.GetDescription()
```

File: tools/addrundata/rundata.go [hunk 7]
```diff
@@ -91,29 +122,22 @@ func (pd *parsedData) parseInit(sc *bufio.Scanner) error {
...
+plan_id: "{{.PlanID}}"
+description: "{{.Description}}"
 `))
 
-// write generates a complete rundata_test.go to the writer.
-func (pd *parsedData) write(w io.Writer, pkg string) error {
-	tmpl.Execute(w, &struct {
-		Package string
-		Data    []struct{ Key, Value string }
+// write generates a complete metadata.textproto to the writer.
```
Comment (liulk) [inline tools/addrundata/rundata.go:134]: Request change: we cannot simply overwrite metadata with only these fields. If there are other fields present in the metadata (e.g. testbed, config and telemetry coverage), we have to preserve them.

**Review by liulk — CHANGES_REQUESTED** [review]

**Comment by greg-dennis** [comment]
Deleting this PR as the new plan splits the work into three different PRs

**Comment by liulk** [comment]
Continued as #1303

**Pull request closed without merging**